app/components/primer/button_group_component.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Primer
+  # Use ButtonGroupComponent to render a series of buttons.
+  class ButtonGroupComponent < Primer::Component
+    include ViewComponent::SlotableV2
+
+    renders_many :buttons, ->(classes: "", **kwargs) do
```

Comment:
From the other PR:

> @joelhawksley: What do you think is the best way to document that these slots take all of the arguments for ButtonComponent?


> @manuelpuyol: That's a good question. I don't think we aren't documenting slots in other components
@srt32 wdyt?

Reply:
@manuelpuyol sounds like it might be time to invent the right solution then 😄 

We can leave this PR as-is in that regard for now.

Reply:
I've seen us document slots v1 behavior like this https://github.com/primer/view_components/pull/182/files#diff-a870f147518fb3db01d7aacb41bc479052f20a68f7e0f934fc78715d1851dd68R36-R39. We'd have to rewrite the Rakefile parser to handle slots v2?

Reply:
since I'm not defining a class as a slot here idk how YARD would deal with it

Reply:
Open an issue for us track figuring this out? No need to block here.